### PR TITLE
WIP Remove immutable from the pause reducer 

### PR DESCRIPTION
**Pull request by bomsy**
Associated Issue: #2283

### Summary of Changes

* Refactor pause reducer to remove immutable

### CODE REVIEW

File: src/utils/scopes.js [hunk 1]
```diff
@@ -23,25 +22,15 @@ function getBindingVariables(bindings, parentName) {
...
+  let thrown = get(pauseInfo, "why.frameFinished.throw", undefined);
 
-  let returned = pauseInfo.getIn(["why", "frameFinished", "return"], undefined);
+  let returned = get(pauseInfo, "why.frameFinished.return", undefined);
 
   const vars = [];
 
   if (thrown !== undefined) {
-    thrown = dehydrateValue(thrown);
+    // thrown = dehydrateValue(thrown);
```
Comment (bomsy) [inline src/utils/scopes.js]: take this out

File: src/utils/scopes.js [hunk 2]
```diff
@@ -50,7 +39,7 @@ export function getSpecialVariables(pauseInfo: Pause, path: string) {
   }
 
   if (returned !== undefined) {
-    returned = dehydrateValue(returned);
+    // returned = dehydrateValue(returned);
```
Comment (bomsy) [inline src/utils/scopes.js]: remove this comment

File: src/utils/scopes.js [hunk 3]
```diff
@@ -79,10 +68,8 @@ function getThisVariable(frame: Frame, path: string) {
   };
 }
 
-export function getScopes(
-  pauseInfo: Pause,
-  selectedFrame: Frame
-): ?(ScopeData[]) {
+export function getScopes(pauseInfo: any, selectedFrame: any): ?(ScopeData[]) {
+  console.log(pauseInfo);
```
Comment (bomsy) [inline src/utils/scopes.js]: forgot this :)

File: flow-typed/debugger-html.js [hunk 4]
```diff
@@ -139,7 +139,6 @@ declare module "debugger-html" {
   declare type Pause = {
     frames: Frame[],
     why: Why,
-    getIn: (string[]) => any,
```
Comment (jasonLaster) [inline flow-typed/debugger-html.js]: lol - good catch

File: src/components/Editor/Preview.js [hunk 5]
```diff
@@ -46,7 +46,7 @@ class Preview extends Component {
       return;
     }
 
-    if (value.actor && !loadedObjects.has(value.actor)) {
+    if (value.actor && !loadedObjects[value.actor]) {
```
Comment (jasonLaster) [inline src/components/Editor/Preview.js]: do you think `hasOwnProperty` would be nice here? it's a tad more explicit

File: src/components/Editor/Preview.js [hunk 6]
```diff
@@ -99,7 +99,7 @@ class Preview extends Component {
   renderObjectInspector(root) {
     const { loadObjectProperties, loadedObjects } = this.props;
 
-    const getObjectProperties = id => loadedObjects.get(id);
+    const getObjectProperties = id => loadedObjects[id];
```
Comment (jasonLaster) [inline src/components/Editor/Preview.js]: this makes me think we might want to use a JS `Map`... We'd have a slightly nicer API

File: src/components/SecondaryPanes/Expressions.js [hunk 7]
```diff
@@ -229,6 +229,16 @@ class Expressions extends PureComponent {
...
 
+Expressions.propTypes = {
+  expressions: PropTypes.array.isRequired,
+  addExpression: PropTypes.func.isRequired,
+  evaluateExpressions: PropTypes.func.isRequired,
+  updateExpression: PropTypes.func.isRequired,
+  deleteExpression: PropTypes.func.isRequired,
+  loadObjectProperties: PropTypes.func,
+  loadedObjects: PropTypes.object.isRequired
+};
```
Comment (jasonLaster) [inline src/components/SecondaryPanes/Expressions.js]: expressions now use `props`

File: src/reducers/pause.js [hunk 8]
```diff
@@ -73,53 +63,62 @@ function update(
...
-      return state.set("selectedFrameId", action.frame.id);
+      return Object.assign({}, state, { selectedFrameId: action.frame.id });
 
     case constants.LOAD_OBJECT_PROPERTIES:
       if (action.status === "start") {
-        return state.setIn(["loadedObjects", action.objectId], {});
+        return {
+          ...state,
+          loadedObjects: {
+            ...state.loadedObjects,
```
Comment (jasonLaster) [inline src/reducers/pause.js]: nice

File: src/reducers/pause.js [hunk 9]
```diff
@@ -73,53 +63,62 @@ function update(
...
+        let pause = frames[0];
 
-        return state.merge({ pause, frames });
+        return Object.assign({}, state, { pause, frames });
       }
 
       break;
     case constants.BREAK_ON_NEXT:
-      return state.set("isWaitingOnBreak", true);
+      return Object.assign({}, state, { isWaitingOnBreak: true });
```
Comment (jasonLaster) [inline src/reducers/pause.js]: I'd like to try keeping this an Immutable map, but just not hydrating the pause data.

I can jump in and look tonight

File: src/utils/scopes.js [hunk 10]
```diff
@@ -158,8 +145,8 @@ export function getScopes(
  * TODO: returns global variables as well
  */
 export function getVisibleVariablesFromScope(
-  pauseInfo: Pause,
-  selectedFrame: Frame
+  pauseInfo: any,
+  selectedFrame: any
```
Comment (jasonLaster) [inline src/utils/scopes.js]: why is this changing?

Comment (bomsy) [inline src/utils/scopes.js]: hmmm..good question!

File: src/reducers/pause.js (hunk 9 above)
Comment (bomsy) [inline src/reducers/pause.js]: kk

**Pull request merged**